frontend/src/utils/index.js

```diff
@@ -777,6 +778,44 @@ export function expiringWorkerGroupsForShoot (shootWorkerGroups, shootCloudProfi
   })
 }
 
+export function convertDuplicateSelectedItemsToPlaceholders (items, sort = true, valuePropertyName) {
```

Comment:
maybe introduce a class to have a frame around the two helper functions and to make it more readable?

class DuplicateSelectedItemsHelper

function itemsToPlaceholders
function placeholdersToItems

Reply:
I reworked the whole code, please have a look. It might be still an option to move this into a class, but as we (currently) do not store property data, this might be over engineered. The data is held by the component that owns it and only uses the functions to convert between wrapped and non-wrapped format.